src/App.tsx

```diff
@@ -4,44 +4,50 @@ import { SubmitLink } from "./components/SubmitLink";
 import { Message } from "./components/Message";
 import { Listing } from "./components/Listing";
 import { ListItem } from "./types/ListItem";
+import "./App.css";
 
-const App = (): JSX.Element => {
+const App: React.FC<Record<string, never>> = () => {
   const submitForm = () => {
     const currentDate = new Date();
-    setUnreadItems([
-      ...unreadItems,
+    setItems([
+      ...items,
       {
-        id: unreadItems.length + 1,
+        id: items.length + 1,
         url: url,
         date: currentDate.toLocaleString(),
+        status: false,
       },
     ]);
     setUrl("");
   };
-  const moveItems = (listItem: ListItem) => {
-    const indexToMove = unreadItems.findIndex((it) => it.id === listItem.id);
-    unreadItems.splice(indexToMove, 1);
-    setUnreadItems([...unreadItems]);
-    setReadItems([...readItems, listItem]);
+  const changeItemStatus = (listItem: ListItem) => {
```

Comment:
This works and is ok to do it this way.
But as we learnt last time, the `.splice(...)` method also allows us to remove and add elements at the same time.
Also it is not ideal to modify the parameter directly, use its copy instead for [immutability reasons](https://en.wikipedia.org/wiki/Immutable_object). In general if there's something that comes to our function / component as a parameter as a prop, it's good not to change it directly. As we talked in react - props flow down and if we want to change them, we tell the component higher in the tree to update them. We should not update those props directly. Same applies to function parameters.

My suggestion here would be to do something like:
`items.splice(items.findindex(...), 1, {...listItem, status: !listItem.status})`
maybe we might need to call `setItems(...)` afterwards as well, not sure tho.

Reply:
Makes absolute sense, great.